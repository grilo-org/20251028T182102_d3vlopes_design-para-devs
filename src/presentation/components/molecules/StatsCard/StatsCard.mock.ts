import { generateID } from '@/utils/generateID';

import { StatsCardProps } from '.';

export const statsMock: StatsCardProps[] = [
	{
		id: generateID(),
		number: '15',
		text: 'Capítulos',
	},
	{
		id: generateID(),
		number: '+250',
		text: 'Páginas',
	},
	{
		id: generateID(),
		number: '3',
		text: 'Bônus',
	},
];
