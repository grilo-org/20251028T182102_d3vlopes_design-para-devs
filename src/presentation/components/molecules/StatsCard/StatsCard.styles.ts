import { styleWrapper } from '@/libs/styles/tailwind-variants';

const styles = styleWrapper({
	slots: {
		wrapperStyles:
			'flex flex-col justify-center items-center w-[11.125rem] min-w-[11.125rem] h-[9.1875rem] bg-background-400 rounded-[4px] md:px-8 md:w-max md:h-[12.9375rem]',
		numberStyles:
			'inline-block text-transparent bg-clip-text bg-gradient-primary font-highlight font-bold text-10xl leading-none md:text-14xl md:leading-none',
		textStyles: 'font-highlight text-lg text-neutral-50',
	},
});

const makeStyles = styles();

export const classNames = {
	wrapperStyles: makeStyles.wrapperStyles(),
	numberStyles: makeStyles.numberStyles(),
	textStyles: makeStyles.textStyles(),
};
