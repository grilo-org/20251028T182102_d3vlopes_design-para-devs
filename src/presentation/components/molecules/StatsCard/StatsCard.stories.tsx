import type { Meta, StoryObj } from '@storybook/react';

import { statsMock as mock } from './StatsCard.mock';

import { StatsCard } from '.';

const meta: Meta<typeof StatsCard> = {
	title: 'Molecules/StatsCard',
	component: StatsCard,
	args: mock[0],
	parameters: {
		design: {
			type: 'figma',
			url: 'https://www.figma.com/file/p1r0o56xHSpG3YHnrNlO3P/LP-Design-para-Devs?type=design&node-id=143-107&mode=design&t=4t5NiTbZSbqSDlzE-0',
		},
	},
};

export default meta;
type Story = StoryObj<typeof StatsCard>;

const template: Story = {
	render: (args) => <StatsCard {...args} />,
};

export const Default: Story = {
	...template,
};
