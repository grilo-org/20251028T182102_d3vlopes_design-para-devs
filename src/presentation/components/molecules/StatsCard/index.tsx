import { StatsModel } from '@/domain/models/Stats';

import { classNames } from './StatsCard.styles';

export type StatsCardProps = StatsModel;

export const StatsCard = ({ number, text }: StatsCardProps) => {
	return (
		<div className={classNames.wrapperStyles}>
			<strong className={classNames.numberStyles}>{number}</strong>
			<span className={classNames.textStyles}>{text}</span>
		</div>
	);
};
