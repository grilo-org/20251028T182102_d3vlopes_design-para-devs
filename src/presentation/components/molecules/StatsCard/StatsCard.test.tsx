import { render, screen } from '@testing-library/react';

import { statsMock } from './StatsCard.mock';

import { StatsCard } from '.';

const { getByText } = screen;

const mock = statsMock[0];

describe('<StatsCard />', () => {
	beforeEach(() => {
		render(<StatsCard {...mock} />);
	});

	it('should render number', () => {
		const statsNumber = getByText(mock.number);

		expect(statsNumber).toBeInTheDocument();
	});

	it('should render text', () => {
		const statsText = getByText(mock.text);

		expect(statsText).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const { container } = render(<StatsCard {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
