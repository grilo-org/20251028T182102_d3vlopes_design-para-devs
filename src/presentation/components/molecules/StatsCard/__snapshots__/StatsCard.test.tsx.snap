// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatsCard /> should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center w-[11.125rem] min-w-[11.125rem] h-[9.1875rem] bg-background-400 rounded-[4px] md:px-8 md:w-max md:h-[12.9375rem]"
  >
    <strong
      class="inline-block text-transparent bg-clip-text bg-gradient-primary font-highlight font-bold text-10xl leading-none md:text-14xl md:leading-none"
    >
      15
    </strong>
    <span
      class="font-highlight text-lg text-neutral-50"
    >
      Cap√≠tulos
    </span>
  </div>
</div>
`;
